From 0d0ca00e56024a9919c150dbed62050d4c70b0c8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?C=C3=A9dric=20Ollivier?= <cedric.ollivier@orange.com>
Date: Wed, 3 Jun 2020 15:23:59 +0200
Subject: [PATCH] Switch to threading.Thread() for Rally tasks
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

multiprocessing.Process() often fails due to thread crashes [1].
It looks similar to gsutil release notes [2].

[1] https://build.opnfv.org/ci/job/functest-opnfv-functest-benchmarking-cntt-latest-rally_full_cntt-run/35/console
[2] https://github.com/GoogleCloudPlatform/gsutil/issues/548
[3] https://github.com/GoogleCloudPlatform/gsutil/blob/master/CHANGES.md

Change-Id: I582933832e23d188c7fa5999e713dd5d7e82d2da
Signed-off-by: Cédric Ollivier <cedric.ollivier@orange.com>
---
 rally/task/runner.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rally/task/runner.py b/rally/task/runner.py
index 61842f1..7d6e2bb 100644
--- a/rally/task/runner.py
+++ b/rally/task/runner.py
@@ -19,6 +19,7 @@ import abc
 import collections
 import copy
 import multiprocessing
+import threading
 import time
 import typing as t
 
@@ -264,7 +265,7 @@ class ScenarioRunner(
                 "processes_to_start": processes_to_start,
                 "processes_counter": i,
             }
-            process = multiprocessing.Process(
+            process = threading.Thread(
                 target=worker_process,
                 args=next(worker_args_gen),
                 kwargs={"info": kwrgs},
-- 
2.26.2
